### PR TITLE
[workflow] update to the latest version 0.10.5

### PR DESCRIPTION
**Pull request by dengjun101**
**Describe the pull request**

- #### What does your PR fix?
  Update workflow to the latest version

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Not Changed

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/workflow/portfile.cmake [hunk 1]
```diff
@@ -2,24 +2,34 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
...
+        DISABLE_PARALLEL_CONFIGURE
+        OPTIONS "-DWORKFLOW_BUILD_STATIC_RUNTIME=ON"
+    )
+else()
+    vcpkg_cmake_configure(
+        SOURCE_PATH "${SOURCE_PATH}"
+        DISABLE_PARALLEL_CONFIGURE
+        OPTIONS "-DWORKFLOW_BUILD_STATIC_RUNTIME=OFF"
+    )
+endif()
```
Comment (Adela0814) [inline ports/workflow/portfile.cmake:31]: You can set a variable for the value of this option to judge in the condition `if(VCPKG_LIBRARY_LINKAGE STREQUAL static)`.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/workflow/portfile.cmake [hunk 2]
```diff
@@ -2,25 +2,31 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
...
         REPO sogou/workflow
-        REF v0.9.7
-        SHA512 4866d9cfe2d9ba30f2f7866819ee8f425b91082d7f86994c1194a6b4406e8ee99e22ce6b0bafeb22c5f098f7da30029fb6b12895c2ac45810d33c28d4bfad006
+        REF v0.10.5
+        SHA512 696e82a1f6a7e6c339fbabb7b1f98ffe40f5f5ee7e77f4c947c0c1532817409e7a61f020c6238a32acd9eb3e06cf3e522e6d67beda32d5bbb08ea1080c20277d
         HEAD_REF master
     )
 endif()
 
+if(VCPKG_CRT_LINKAGE STREQUAL "static")
```
Comment (Adela0814) [inline ports/workflow/portfile.cmake:19]: if (VCPKG_LIBRARY_LINKAGE STREQUAL static)

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/workflow/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/workflow/portfile.cmake:19]: NO NO NO.

Comment (dg0yt) [reply ports/workflow/portfile.cmake:19]: It is about runtime, isn't it?
And we want quotes around `static`.

**Review by Adela0814 — DISMISSED** [review]

File: ports/workflow/portfile.cmake [hunk 3]
```diff
@@ -2,25 +2,31 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
...
+    set(CONFIGURE_OPTIONS "-DWORKFLOW_BUILD_STATIC_RUNTIME=ON")
+else()
+    set(CONFIGURE_OPTIONS "-DWORKFLOW_BUILD_STATIC_RUNTIME=OFF")
+endif()
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     DISABLE_PARALLEL_CONFIGURE
+    OPTIONS "${CONFIGURE_OPTIONS}"
```
Comment (BillyONeal) [inline ports/workflow/portfile.cmake:28]:
```suggestion
    OPTIONS ${CONFIGURE_OPTIONS}
```

The name being plural suggests that this is intended to be expanded like a list but you are putting it in quotes here. Is that the intent?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

**Comment by dengjunplusplus** [comment]
@BillyONeal PTAL

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**